src/common/string_util.cc

```diff
@@ -276,6 +276,71 @@ std::pair<std::string, std::string> SplitGlob(std::string_view glob) {
   return {prefix, ""};
 }
 
+StatusOr<std::vector<std::string>> SplitArguments(std::string_view in) {
+  std::vector<std::string> arguments;
+  std::string current_string;
+
+  enum State { NORMAL, DOUBLE_QUOTED, SINGLE_QUOTED, ESCAPE } state = NORMAL;
+
+  State state_before_escape = NORMAL;
+  for (const char c : in) {
+    switch (state) {
+      case NORMAL:
+        if (std::isspace(c)) {
+          if (!current_string.empty()) {
+            arguments.emplace_back(std::move(current_string));
+            current_string.clear();
+          }
+          // skip spaces
+        } else if (c == '"' || c == '\'') {
+          state = c == '"' ? DOUBLE_QUOTED : SINGLE_QUOTED;
+        } else {
+          current_string.push_back(c);
+        }
+        break;
+      case DOUBLE_QUOTED:
+      case SINGLE_QUOTED:
+        if (c == '\\') {
+          state_before_escape = state;
+          state = ESCAPE;
+        } else if ((c == '"' && state == DOUBLE_QUOTED) || (c == '\'' && state == SINGLE_QUOTED)) {
+          state = NORMAL;
```

Comment:
should we checks that:

```
 /* closing quote must be followed by a space or
                         * nothing at all. */
```

Reply:
Yeah, different behavior in `a "b"c d` with redis here.